core/file_handler.py

```
@@ -23,3 +25,13 @@ def receive_data_chunk(self, raw_data, start):
             if mime not in self.ACCEPTED_FILE_UPLOAD_MIME_TYPES:
                 self.abort(PermissionDenied("Unsupported file type"))
         super().receive_data_chunk(raw_data, start)
+
+    def file_complete(self, *args, **kwargs):
+        super().file_complete(*args, **kwargs)
+
+        self.file = S3Boto3StorageFile(self.s3_key, 'r', self.storage)
+        self.file.original_name = self.file_name
+        self.file.content_type = self.content_type
+        self.file.charset = None
+
+        return self.file
```

This probably needs some comments to explain why I did this in the first place.

I'm also not completely sure that `charset = None` is the right thing to do, it needs to be set to something though but not sure if `None` is the right thing.

---

Thanks, yes comment added. I've also propagated the charset rather than explicitly setting it to `None` which seems to work fine, e.g.
```python
self.file.charset = self.charset
```